### PR TITLE
[onert] Introduce UseDefChain

### PR DESCRIPTION
**Pull request by ragmani**
This commit introduces UseDefChain that contains use-def edges between operands and operartions.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

**Review by hseok-oh — DISMISSED** [review]
LGTM

File: runtime/onert/core/include/ir/train/UseDefChains.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __ONERT_IR_TRAIN_USEDEFCHAINS_H__
+#define __ONERT_IR_TRAIN_USEDEFCHAINS_H__
+
+#include "ir/train/UseDefChain.h"
+#include "ir/train/Index.h"
+#include "util/ObjectManager.h"
```
Comment (zetwhite) [inline runtime/onert/core/include/ir/train/UseDefChains.h:22]: Is this header necessary?

File: runtime/onert/core/src/ir/train/UseDefChain.cc [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+void UseDefChain::clearTrainingUseDefs()
+{
+  _uses.clear();
+  _defs.clear();
+}
+
+bool UseDefChain::operator==(const UseDefChain &other) const
+{
+  return _uses == other._uses && _defs == other._defs;
```
Comment (zetwhite) [inline runtime/onert/core/src/ir/train/UseDefChain.cc:50]: Did you intentionally avoid comparing `_operand`?

**Review by zetwhite — DISMISSED** [review]
My questions are minor things.  
Overall codes looks good to me.

File: runtime/onert/core/include/ir/train/UseDefChains.h (hunk 1 above)
Comment (ragmani) [reply runtime/onert/core/include/ir/train/UseDefChains.h:22]: You are right. It is a legacy left over from revision in the draft.

File: runtime/onert/core/src/ir/train/UseDefChain.cc (hunk 2 above)
Comment (ragmani) [reply runtime/onert/core/src/ir/train/UseDefChain.cc:50]: I missed it. I think it's better to add comparing `_operand` as well.

**Review by hseok-oh — APPROVED** [review]

**Review by zetwhite — APPROVED** [review]

**Pull request merged**